src/lib/mina_net2/mina_net2.ml

```diff
@@ -1,1367 +1,116 @@
-    (* The remote peer closed its write end of one of our streams *)
-    | "streamReadComplete" -> (
-        let%bind m = Stream_read_complete.of_yojson v |> or_error in
-        let stream_idx = m.stream_idx in
-        match Hashtbl.find t.streams stream_idx with
-        | Some stream ->
-            advance_stream_state t stream `Them |> don't_wait_for ;
-            Ok ()
-        | None ->
-            Or_error.errorf
-              "streamReadComplete for stream we don't know about %d" stream_idx
-        )
-    | s ->
-        Or_error.errorf "unknown upcall %s" s
+  let of_peer ({ libp2p_port; host; _ } : Peer.t) =
+    { libp2p_port; host = Unix.Inet_addr.to_string host }
 end
-(* Warning 30 is about field labels being defined in multiple types.
-   It means more disambiguation has to happen sometimes but it doesn't
-   seem to be a big deal. *)
-[@warning "-30"]
-
-type net = Helper.t
 
-module Keypair = struct
-  include Keypair0
-
-  let random net =
-    match%map Helper.do_rpc net (module Helper.Rpcs.Generate_keypair) () with
-    | Ok { sk; pk; peer_id } ->
-        (let open Or_error.Let_syntax in
-        let%bind secret = of_b64_data (`String sk) in
-        let%map public = of_b64_data (`String pk) in
-        ({ secret; public; peer_id = Peer.Id.unsafe_of_string peer_id } : t))
-        |> Or_error.ok_exn
-    | Error e ->
-        Error.tag e ~tag:"Other RPC error generateKeypair" |> Error.raise
-
-  let secret_key_base64 ({ secret; _ } : t) = to_b64_data secret
-
-  let to_string ({ secret; public; peer_id } : t) =
-    String.concat ~sep:","
-      [ to_b64_data secret; to_b64_data public; Peer.Id.to_string peer_id ]
-
-  let of_string s =
-    let parse_with_sep sep =
-      match String.split s ~on:sep with
-      | [ secret_b64; public_b64; peer_id ] ->
-          let open Or_error.Let_syntax in
-          let%map secret = of_b64_data (`String secret_b64)
-          and public = of_b64_data (`String public_b64) in
-          ({ secret; public; peer_id = Peer.Id.unsafe_of_string peer_id } : t)
-      | _ ->
-          Or_error.errorf "%s is not a valid Keypair.to_string output" s
-    in
-    let with_semicolon = parse_with_sep ';' in
-    let with_comma = parse_with_sep ',' in
-    if Or_error.is_error with_semicolon then with_comma else with_semicolon
-
-  let to_peer_id ({ peer_id; _ } : t) = peer_id
-end
-
-module Multiaddr = struct
-  type t = string [@@deriving compare, bin_io_unversioned]
-
-  let to_string t = t
-
-  let of_string t = t
-
-  let to_peer t =
-    match String.split ~on:'/' t with
-    | [ ""; "ip4"; ip4_str; "tcp"; tcp_str; "p2p"; peer_id ] -> (
-        try
-          let host = Unix.Inet_addr.of_string ip4_str in
-          let libp2p_port = Int.of_string tcp_str in
-          Some (Network_peer.Peer.create host ~libp2p_port ~peer_id)
-        with _ -> None )
-    | _ ->
-        None
+(* TODO: connection gating info is currently stored in to places, that needs to be fixed... *)
```

Comment:
what's the other place, just for my benefit?

Reply:
We store 2 copies of the connection_gating struct in different layers of our networking code.

https://github.com/MinaProtocol/mina/pull/9472/files/c78dd445543cc3f6c3d6f69ab2a8d65bb2e82615#diff-319cec3b56ef1fc2d50525ad5239c6dac7563a2de7d3739efd7af6fda2acbc1fR71

https://github.com/MinaProtocol/mina/pull/9472/files/c78dd445543cc3f6c3d6f69ab2a8d65bb2e82615#diff-ba0e0e31ca94b59ea8bec503e1358c3037c612a7b0bba8d400b9450bfbc30f30L507